pkg/commands/compute/serve.go

```diff
@@ -648,6 +648,25 @@ func local(opts localOpts) error {
 	// we do finally come to stop the `serve` command (e.g. user presses Ctrl-c).
 	// How big an issue this is depends on how many file modifications a user
 	// makes, because having lots of signal listeners could exhaust resources.
+	//
+	// When there is an error setting up the watching of files, if we error we
+	// need to signal the error using a channel as watching files happens
+	// asynchronously in a goroutine. We also need to be able to signal the
+	// viceroy process to be killed, and we do that using `s.Signal(os.Kill)` from
+	// within the relevant error handling blocks in `watchFiles`, where upon the
+	// below `select` statement will pull the error message from the `failure`
+	// channel and return it to the user. If we fail to kill the Viceroy process
+	// then we still want to pull an error from the `failure` channel and so we
+	// have a separate `select` statement to check for any initial errors prior to
+	// the Viceroy executable starting and an error occurring in `watchFiles`.
+	select {
+	case channelErr := <-failure:
```

Comment:
What would you think of renaming these 2 variables (here and elsewhere)? Looking only here, it's not clear what a receive on the channel indicates. Maybe something like `fileWatchErrs`? Likewise for `channelErr`, maybe `fileWatchErr`?

Reply:
Ah so I actually purposefully chose generic naming because although it's handling file watch errors right now, maybe in the future I want other types of errors to cause the stopping of the process.

Reply:
I could maybe name it something like `asyncErrs := <-failure`

Reply:
Done